### ISSUE TITLE
cmd/go: tries to download v2.2.2 (no suffix) when checking a replaced v2.2.2+incompatible for a missing import

### ISSUE THREAD

**Issue by leighmcculloch** [golang/go#33795]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13rc1 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN="/home/leighmcculloch/local/bin"
GOCACHE="/home/leighmcculloch/.cache/go-build"
GOENV="/home/leighmcculloch/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/leighmcculloch/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/leighmcculloch/local/bin/go/1.13rc1"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/leighmcculloch/local/bin/go/1.13rc1/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/leighmcculloch/devel/test/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build816830480=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

I run `go mod tidy` on the following code.

`go.mod`:
```
module test

go 1.13

require (
        github.com/throttled/throttled v2.2.2+incompatible
)

replace github.com/throttled/throttled v2.2.2+incompatible => github.com/bartekn/throttled v2.2.3-0.20181102161246-c99eef3ad70a+incompatible
```

`main.go`:
```go
package main

import (
        "fmt"

        "github.com/throttled/throttled"
)

func main() {
        fmt.Println(throttled.HTTPRateLimiter{})
}
```

### What did you expect to see?

I expected `go mod tidy` to exit status code of success (`0`).

```
$ go mod tidy
$ echo $?
0
```

### What did you see instead?

The `go mod tidy` command had a failure exit status code (`1`) and a `410 Gone` error message.

```
$ go mod tidy
go: downloading github.com/throttled/throttled v2.2.2
verifying github.com/throttled/throttled@v2.2.2: github.com/throttled/throttled@v2.2.2: reading https://sum.golang.org/lookup/github.com/throttled/throttled@v2.2.2: 410 Gone
$ echo $?
1
```

If I visit the URL https://sum.golang.org/lookup/github.com/throttled/throttled@v2.2.2 it returns the text:

```
not found: invalid version
```

If I remove the `replace` directive, `go mod tidy` returns success and there is no verifying error message. The output in the section above where the exit status is `0` is what I see when the `replace` statement is removed. The problem I'm experiencing is isolated to when the `replace` directive is in use.

The version of the throttled package is `v2.2.2+incompatible` and so if the tool wants to query the sumdb about it, it should be attempting to query the URL https://sum.golang.org/lookup/github.com/throttled/throttled@v2.2.2+incompatible, but for some reason it is attempting to go to the URL without the `+incompatible` suffix.

It also seems strange that it is attempting to download the source package when I have a replace directive present pointing to the fork. Possibly it shouldn't be doing that either?

### Context

The above example is a reduced example of an error I'm seeing while I'm trying to migrate a larger Go repo from dep to Modules (stellar/go#1634).

**Comment by leighmcculloch** [comment]
I tried this on Go 1.12.9, and the `go mod tidy` command works as expected there, so this only seems to be broken on Go 1.13rc1. I assume that's because in 1.12 there is no call out to the sumdb.

**Comment by bcmills** [comment]
Found the bug. It's another variation on #32700.

**Comment by bcmills** [comment]
With the fix, I get a successful `go mod tidy` with upgraded versions:
```
test$ go mod tidy
go: downloading github.com/throttled/throttled v1.0.0
go: downloading github.com/throttled/throttled v2.2.4+incompatible
go: finding versions for github.com/throttled
go: finding versions for github.com/throttled/throttled/store
go: finding versions for github.com/throttled/throttled/store/memstore
go: found github.com/throttled/throttled/store in github.com/throttled/throttled v2.2.4+incompatible
go: downloading github.com/gomodule/redigo v1.7.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: found github.com/gomodule/redigo/redis in github.com/gomodule/redigo v2.0.0+incompatible

test$
```

**Comment by bcmills** [comment]
This seems like a small and obvious enough fix to make Go 1.14. The bug has been present since [CL 152739](https://golang.org/cl/152739), but was masked by the lack of error reporting in `go mod tidy` in 1.12.
